Dockerfile

```diff
@@ -1,23 +1,30 @@
-FROM openjdk:17-jdk@sha256:528707081fdb9562eb819128a9f85ae7fe000e2fbaeaf9f87662e7b3f38cb7d8 as build
+# build commons with java 21
+FROM amazoncorretto:21-alpine@sha256:937a7f5c5f7ec41315f1c7238fd9ec0347684d6d99e086db81201ca21d1f5778 AS commons-builder
+
+WORKDIR /workspace/app
+RUN apk add --no-cache git findutils
+
+COPY . .
+RUN chmod +x ./gradlew
+RUN chmod +x ./pagopa-ecommerce-commons-maven-install.sh
+RUN ./gradlew install-commons -PbuildCommons
```

Comment:
RUN ./gradlew build -x test -PbuildCommons perform in a single shot commons + service build, sharing the same .m2 cache. why splitting this into two build stages?

Reply:
eCommerce commons library dependency should bu upgraded manually on lock file unforunately.
This is needed since this library is built on the fly, so it's sha value cannot be calculated at-priori (changing built timestamp into manifest file).
Since it's an internally developed library and checkout is performed from tags it's safe to let it un-verified